lib/rom/csv/dataset/write_operations.rb

```diff
@@ -0,0 +1,43 @@
+module ROM
+  module CSV
+    class Dataset
+      module WriteOperations
+        def add_headers(headers)
+          @headers = headers
+        end
+
+        def <<(tuples)
+          if tuples.is_a?(Array)
+            tuples.each { |tuple| data << tuple }
+          else
+            data << tuples
+          end
+        end
+
+        def update(index, tuple)
+          data[index].merge!(tuple)
+        end
+
+        def delete(tuple)
+          data.delete(tuple)
+        end
+
+        def write
+          ::CSV.open(path, 'wb') do |csv|
```

Comment:
We should pass `options` passed via `ROM.setup(:csv, "file", options)` to `CSV.open`.